setup.py

```diff
@@ -124,29 +124,18 @@
     "networkx.utils": ["tests/*.py"],
 }
 
-install_requires = ["decorator>=4.3.0"]
+
+def parse_requirements_file(filename):
+    with open(filename, encoding="utf-8") as fid:
+        requires = [l.strip() for l in fid.readlines() if l]
+
+    return requires
+
+
+install_requires = parse_requirements_file("requirements/default.txt")
 extras_require = {
-    "all": [
-        "numpy",
-        "scipy",
-        "pandas",
-        "matplotlib",
-        "pygraphviz",
-        "pydot",
-        "pyyaml",
-        "lxml",
-        "pytest",
-    ],
-    "gdal": ["gdal"],
-    "lxml": ["lxml"],
-    "matplotlib": ["matplotlib"],
-    "pytest": ["pytest"],
-    "numpy": ["numpy"],
-    "pandas": ["pandas"],
-    "pydot": ["pydot"],
-    "pygraphviz": ["pygraphviz"],
-    "pyyaml": ["pyyaml"],
-    "scipy": ["scipy"],
+    dep: parse_requirements_file("requirements/" + dep + ".txt")
+    for dep in ["extra", "test"]
```

Comment:
Do we want to add `"doc"` to this list?

Reply:
I was imaging that setup.py was mostly for end-users and that developers would continue to use the requirements files.  I assume most users are not going to be building the documentation for themselves.  And if they are building the documentation, they should probably be working from a git clone of the repo.

Reply:
True, though on the other hand I don't think it necessarily clutters anything up to have the other dependencies here. I'm probably biased because I use `pip` to set up environments for development workflows, so to set up a complete environment would simply be `pip install -e .[extra,test,doc]` instead of having to explicitly call `pip -r`. Not a blocker for me though, it's fine the way it is!

Reply:
Now you can do ``pip install -e .[developer,doc,extra,test]``.